app/src/main/java/chat/rocket/android/chatroom/ui/PinnedMessagesFragment.kt

```diff
@@ -88,9 +88,22 @@ class PinnedMessagesFragment : Fragment(), PinnedMessagesView {
                         }
                     })
                 }
+                togglePinView(pinnedMessages.size)
             }
 
             adapter.appendData(pinnedMessages)
         }
     }
+
+    private fun togglePinView(size: Int) {
+     if (size == 0){
+         iv_pin_icon.visibility = View.VISIBLE
+         tv_pin_title.visibility = View.VISIBLE
+         tv_pin_description.visibility = View.VISIBLE
+     }else{
+         iv_pin_icon.visibility = View.GONE
```

Comment:
Change `iv_pin_icon.visibility = View.VISIBLE` to `iv_pin_icon.setVisible(false)` - as well as for the others.